elfi/testbench/testbench.py

```diff
@@ -0,0 +1,295 @@
+
+            if self.progress_bar:
+                self.progress_bar.reinit_progressbar(reinit_msg=method.attributes['name'])
+
+            self.testbench_results.append(
+                self._repeat_inference(method, self.method_seed_list[method_index])
+                )
+
+    def _repeat_inference(self, method, seed_list):
+        repeated_result = []
+        model = self.model.copy()
+        for i in np.arange(self.repetitions):
+            if self.progress_bar:
+                self.progress_bar.update_progressbar(i + 1, self.repetitions)
+
+            model.observed[self.simulator_name] = self.observations[i]
+
+            repeated_result.append(
+                self._draw_posterior_sample(method, model, seed_list[i])
+                )
+
+        return self._collect_results(
+            method.attributes['name'],
+            repeated_result)
+
+    def _draw_posterior_sample(self, method, model, seed):
+        method_instance = method.attributes['callable'](
+            model,
+            **method.attributes['method_kwargs'],
+            seed=seed)
+
+        fit_kwargs = method.attributes['fit_kwargs']
+
+        if len(fit_kwargs) > 0:
+            method_instance.fit(fit_kwargs)
+
+        sampler_kwargs = method.attributes['sample_kwargs']
+
+        return method_instance.sample(**sampler_kwargs)
+
+    def _collect_results(self, name, results):
+        result_dictionary = {
+            'method': name,
+            'results': results
+        }
+        return result_dictionary
+
+    # TODO
+    def _compare_sample_results(self):
+        """Compare results in sample-format."""
+
+    # TODO
+    def _retrodiction(self):
+        """Infer a problem with known parameter values."""
+
+    def get_testbench_results(self):
+        """Return Testbench testcases and results."""
+        testbench_data = {
+            'testcases': self.test_dictionary,
+            'results': self.testbench_results
+        }
+        return testbench_data
+
+    def collect_RMSE(self):
+        """Collect RMSEs for sample mean for methods included in Testbench."""
+        method_name = []
+        sample_mean_RMSE = []
+        for _, method_results in enumerate(self.testbench_results):
+            method_name.append(method_results['method'])
+            sample_mean_RMSE.append(self._get_squared_sample_mean_difference(method_results))
+
+        RMSE_results = {
+            'method': method_name,
+            'sample_mean_RMSE': sample_mean_RMSE
+        }
+
+        return RMSE_results
+
+    def _get_squared_sample_mean_difference(self, method):
+        RMSE = np.zeros(len(method['results']))
+        for repetition, results in enumerate(method['results']):
+            for keys, values in results.sample_means.items():
+                RMSE[repetition] += (
+                    values - self.reference_parameter[keys][repetition]) ** 2
+
+            RMSE[repetition] += 1 / self.param_dim
```

Comment:
operation on the last line here looks like an error. however i also think that RMSE should be calculated parameterwise and over repetitions,  ie as square root over the expected squared error in each parameter, so the user could see the expected estimation error in each model parameter

Reply:
It is. I changed the example result_get-type function to return method- and parameterwise differences to the reference-parameter-value. This is supposed to only act as an example at this stage as we'll hopefully very soon add some methods for displaying various types of results.